backend/test_observer/controllers/artefacts/logic.py

```diff
@@ -0,0 +1,119 @@
+from collections.abc import Iterable, Iterator
+
+from sqlalchemy.orm import Session, joinedload
+
+from test_observer.data_access.models import ArtefactBuild, TestExecution
+from test_observer.external_apis.c3.models import (
+    Report,
+    SubmissionProcessingStatus,
+    SubmissionStatus,
+)
+
+from .models import (
+    ArtefactBuildDTO,
+    EnvironmentDTO,
+    TestExecutionDTO,
+    TestExecutionStatus,
+)
+
+
+class TestExecutionStatusLogic:
+    NO_C3_LINK = TestExecutionStatus.IN_PROGRESS
+    NO_STATUS_ID_IN_C3_LINK = TestExecutionStatus.UNKNOWN
+    SUBMISSION_STATUS_NOT_FOUND = TestExecutionStatus.UNKNOWN
+    SUBMISSION_STATUS_IS_FAIL = TestExecutionStatus.FAILED
+    NO_REPORT_ID_IN_SUBMISSION_STATUS = TestExecutionStatus.UNKNOWN
+    REPORT_NOT_FOUND = TestExecutionStatus.UNKNOWN
+    NO_FAILED_TESTS = TestExecutionStatus.PASSED
+    SOME_FAILED_TESTS = TestExecutionStatus.FAILED
```

Comment:
should it be an enum?

Reply:
That's what I did initially. But I started getting linting errors or mypy (I forgot which one) that stated you can't have an enum whose value type is an enum too